# HG changeset patch
# User HackaLittleBit
# Date 1342540880 -3600
# Node ID 105cb37f00581ce6edb9e83e08d3e9cb41fe37f1
# Parent  a048e90e006e8627c57dd7df95a7851dbd9de1e3
All changes in vars, except terraform_gui
game_window.hpp.sq manually changed!!!
ShowCreateScenario has to be renamed or removed.

diff -r a048e90e006e -r 105cb37f0058 src/genworld.h
--- a/src/genworld.h	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/genworld.h	Tue Jul 17 17:01:20 2012 +0100
@@ -87,7 +87,9 @@
 void PrepareGenerateWorldProgress();
 void ShowGenerateWorldProgress();
 void StartNewGameWithoutGUI(uint seed);
+void ShowGenerateLandscape();
 void ShowCreateScenario();
+void ShowHeightmapLoad();
 void StartScenarioEditor();
 
 extern bool _generating_world;
diff -r a048e90e006e -r 105cb37f0058 src/lang/english.txt
--- a/src/lang/english.txt	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/lang/english.txt	Tue Jul 17 17:01:20 2012 +0100
@@ -1042,10 +1042,10 @@
 STR_NONE                                                        :None
 STR_FUNDING_ONLY                                                :Funding only
 STR_MINIMAL                                                     :Minimal
-STR_NUM_VERY_LOW                                                :Very Low
-STR_NUM_LOW                                                     :Low
+STR_NUM_VERY_LOW                                                :Very Few
+STR_NUM_LOW                                                     :Few
 STR_NUM_NORMAL                                                  :Normal
-STR_NUM_HIGH                                                    :High
+STR_NUM_HIGH                                                    :Many
 STR_NUM_CUSTOM                                                  :Custom
 STR_NUM_CUSTOM_NUMBER                                           :Custom ({NUM})
 
@@ -1062,10 +1062,10 @@
 STR_AI_SPEED_FAST                                               :Fast
 STR_AI_SPEED_VERY_FAST                                          :Very Fast
 
-STR_SEA_LEVEL_VERY_LOW                                          :Very Low
-STR_SEA_LEVEL_LOW                                               :Low
-STR_SEA_LEVEL_MEDIUM                                            :Medium
-STR_SEA_LEVEL_HIGH                                              :High
+STR_SEA_LEVEL_VERY_LOW                                          :Very Little
+STR_SEA_LEVEL_LOW                                               :Little
+STR_SEA_LEVEL_MEDIUM                                            :Normal
+STR_SEA_LEVEL_HIGH                                              :Much
 STR_SEA_LEVEL_CUSTOM                                            :Custom
 STR_SEA_LEVEL_CUSTOM_PERCENTAGE                                 :Custom ({NUM}%)
 
@@ -1083,6 +1083,7 @@
 STR_SUBSIDY_X3                                                  :x3
 STR_SUBSIDY_X4                                                  :x4
 
+STR_TERRAIN_TYPE_TOTALY_FLAT                                    :Totally Flat
 STR_TERRAIN_TYPE_VERY_FLAT                                      :Very Flat
 STR_TERRAIN_TYPE_FLAT                                           :Flat
 STR_TERRAIN_TYPE_HILLY                                          :Hilly
@@ -1255,7 +1256,7 @@
 
 STR_CONFIG_SETTING_LAND_GENERATOR                               :Land generator: {STRING2}
 STR_CONFIG_SETTING_LAND_GENERATOR_ORIGINAL                      :Original
-STR_CONFIG_SETTING_LAND_GENERATOR_TERRA_GENESIS                 :TerraGenesis
+STR_CONFIG_SETTING_LAND_GENERATOR_TERRA_GENESIS                 :Improved
 STR_CONFIG_SETTING_OIL_REF_EDGE_DISTANCE                        :Max distance from edge for Oil Refineries: {STRING2}
 STR_CONFIG_SETTING_OIL_REF_EDGE_DISTANCE_HELPTEXT               :Oil refineries are only constructed near the map border, that is at the coast for island maps
 STR_CONFIG_SETTING_SNOWLINE_HEIGHT                              :Snow line height: {STRING2}
@@ -1266,8 +1267,8 @@
 STR_CONFIG_SETTING_ROUGHNESS_OF_TERRAIN_VERY_ROUGH              :Very Rough
 STR_CONFIG_SETTING_TREE_PLACER                                  :Tree placer algorithm: {STRING2}
 STR_CONFIG_SETTING_TREE_PLACER_NONE                             :None
-STR_CONFIG_SETTING_TREE_PLACER_ORIGINAL                         :Original
-STR_CONFIG_SETTING_TREE_PLACER_IMPROVED                         :Improved
+STR_CONFIG_SETTING_TREE_PLACER_ORIGINAL                         :Normal
+STR_CONFIG_SETTING_TREE_PLACER_IMPROVED                         :More
 STR_CONFIG_SETTING_HEIGHTMAP_ROTATION                           :Heightmap rotation: {STRING2}
 STR_CONFIG_SETTING_HEIGHTMAP_ROTATION_COUNTER_CLOCKWISE         :Counter clockwise
 STR_CONFIG_SETTING_HEIGHTMAP_ROTATION_CLOCKWISE                 :Clockwise
@@ -2512,29 +2513,33 @@
 
 # World generation
 STR_MAPGEN_WORLD_GENERATION_CAPTION                             :{WHITE}World Generation
-STR_MAPGEN_MAPSIZE                                              :{BLACK}Map size:
-STR_MAPGEN_BY                                                   :{BLACK}*
-STR_MAPGEN_NUMBER_OF_TOWNS                                      :{BLACK}No. of towns:
-STR_MAPGEN_DATE                                                 :{BLACK}Date:
-STR_MAPGEN_NUMBER_OF_INDUSTRIES                                 :{BLACK}No. of industries:
-STR_MAPGEN_SNOW_LINE_HEIGHT                                     :{BLACK}Snow line height:
-STR_MAPGEN_SNOW_LINE_UP                                         :{BLACK}Move the snow line height one up
-STR_MAPGEN_SNOW_LINE_DOWN                                       :{BLACK}Move the snow line height one down
-STR_MAPGEN_RANDOM_SEED                                          :{BLACK}Random seed:
-STR_MAPGEN_RANDOM_SEED_HELP                                     :{BLACK}Click to enter a random seed
+STR_MAPGEN_RANDOM_GAME                                          :{BLACK}Random Game
+STR_MAPGEN_LOAD_SCENARIO                                        :{BLACK}Load Scenario
+STR_MAPGEN_LOAD_HEIGHTMAP                                       :{BLACK}Load Heightmap
+STR_MAPGEN_DATE                                                 :{BLACK}Starting date:
+STR_MAPGEN_ADVANCED_PLUS                                        :{BLACK}Advanced +
+STR_MAPGEN_ADVANCED_MIN                                         :{BLACK}Advanced -
+STR_MAPGEN_MAP                                                  :{GREEN}Map
+STR_MAPGEN_MAPSIZE                                              :{BLACK}Size:
+STR_MAPGEN_BY                                                   :{BLACK}x
+STR_MAPGEN_TERRAIN_TYPE                                         :{BLACK}Terrain type:
+STR_MAPGEN_VARIETY                                              :{BLACK}Land variation:
+STR_MAPGEN_SMOOTHNESS                                           :{BLACK}Smoothness:
+STR_MAPGEN_LAND_GENERATOR                                       :{BLACK}Generator:
+STR_MAPGEN_QUANTITY_OF_SEA_LAKES                                :{BLACK}Sea:
+STR_MAPGEN_SCENARIO                                             :{GREEN}Scenery
+STR_MAPGEN_QUANTITY_OF_RIVERS                                   :{BLACK}Rivers:
+STR_MAPGEN_TREE_PLACER                                          :{BLACK}Trees:
+STR_MAPGEN_NUMBER_OF_TOWNS                                      :{BLACK}Towns:
+STR_MAPGEN_NUMBER_OF_INDUSTRIES                                 :{BLACK}Industries:
+STR_MAPGEN_SNOW_LINE_HEIGHT                                     :{BLACK}Snow line:
+STR_MAPGEN_SNOW_LINE_UP                                         :{BLACK}Move the snow line one tile up
+STR_MAPGEN_SNOW_LINE_DOWN                                       :{BLACK}Move the snow line one tile down
 STR_MAPGEN_RANDOM                                               :{BLACK}Randomise
-STR_MAPGEN_RANDOM_HELP                                          :{BLACK}Change the random seed used for Terrain Generation
-STR_MAPGEN_LAND_GENERATOR                                       :{BLACK}Land generator:
-STR_MAPGEN_TREE_PLACER                                          :{BLACK}Tree algorithm:
-STR_MAPGEN_TERRAIN_TYPE                                         :{BLACK}Terrain type:
-STR_MAPGEN_QUANTITY_OF_SEA_LAKES                                :{BLACK}Sea level:
-STR_MAPGEN_QUANTITY_OF_RIVERS                                   :{BLACK}Rivers:
-STR_MAPGEN_SMOOTHNESS                                           :{BLACK}Smoothness:
-STR_MAPGEN_VARIETY                                              :{BLACK}Variety distribution:
 STR_MAPGEN_GENERATE                                             :{WHITE}Generate
 
 # Strings for map borders at game generation
-STR_MAPGEN_BORDER_TYPE                                          :{BLACK}Map edges:
+STR_MAPGEN_BORDER_TYPE                                          :{BLACK}Edges:
 STR_MAPGEN_NORTHWEST                                            :{BLACK}Northwest
 STR_MAPGEN_NORTHEAST                                            :{BLACK}Northeast
 STR_MAPGEN_SOUTHEAST                                            :{BLACK}Southeast
@@ -2545,8 +2550,9 @@
 STR_MAPGEN_BORDER_RANDOMIZE                                     :{BLACK}Random
 STR_MAPGEN_BORDER_MANUAL                                        :{BLACK}Manual
 
-STR_MAPGEN_HEIGHTMAP_ROTATION                                   :{BLACK}Heightmap rotation:
-STR_MAPGEN_HEIGHTMAP_NAME                                       :{BLACK}Heightmap name:
+STR_MAPGEN_HEIGHTMAP                                            :{GREEN}Heightmap
+STR_MAPGEN_HEIGHTMAP_ROTATION                                   :{BLACK}rotation:
+STR_MAPGEN_HEIGHTMAP_NAME                                       :{BLACK}Name:
 STR_MAPGEN_HEIGHTMAP_SIZE_LABEL                                 :{BLACK}Size:
 STR_MAPGEN_HEIGHTMAP_SIZE                                       :{ORANGE}{NUM} x {NUM}
 
diff -r a048e90e006e -r 105cb37f0058 src/script/api/game/game_window.hpp.sq
--- a/src/script/api/game/game_window.hpp.sq	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/script/api/game/game_window.hpp.sq	Tue Jul 17 17:01:20 2012 +0100
@@ -453,8 +453,6 @@
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_MAPSIZE_Y_PULLDOWN,                 "WID_GL_MAPSIZE_Y_PULLDOWN");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_TOWN_PULLDOWN,                      "WID_GL_TOWN_PULLDOWN");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_INDUSTRY_PULLDOWN,                  "WID_GL_INDUSTRY_PULLDOWN");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_RANDOM_EDITBOX,                     "WID_GL_RANDOM_EDITBOX");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_RANDOM_BUTTON,                      "WID_GL_RANDOM_BUTTON");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_GENERATE_BUTTON,                    "WID_GL_GENERATE_BUTTON");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_START_DATE_DOWN,                    "WID_GL_START_DATE_DOWN");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_START_DATE_TEXT,                    "WID_GL_START_DATE_TEXT");
@@ -477,20 +475,9 @@
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_WATER_NE,                           "WID_GL_WATER_NE");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_WATER_SE,                           "WID_GL_WATER_SE");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_WATER_SW,                           "WID_GL_WATER_SW");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_TEMPERATE,                          "WID_CS_TEMPERATE");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_ARCTIC,                             "WID_CS_ARCTIC");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_TROPICAL,                           "WID_CS_TROPICAL");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_TOYLAND,                            "WID_CS_TOYLAND");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_EMPTY_WORLD,                        "WID_CS_EMPTY_WORLD");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_RANDOM_WORLD,                       "WID_CS_RANDOM_WORLD");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_MAPSIZE_X_PULLDOWN,                 "WID_CS_MAPSIZE_X_PULLDOWN");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_MAPSIZE_Y_PULLDOWN,                 "WID_CS_MAPSIZE_Y_PULLDOWN");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_START_DATE_DOWN,                    "WID_CS_START_DATE_DOWN");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_START_DATE_TEXT,                    "WID_CS_START_DATE_TEXT");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_START_DATE_UP,                      "WID_CS_START_DATE_UP");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_FLAT_LAND_HEIGHT_DOWN,              "WID_CS_FLAT_LAND_HEIGHT_DOWN");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_FLAT_LAND_HEIGHT_TEXT,              "WID_CS_FLAT_LAND_HEIGHT_TEXT");
-	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_CS_FLAT_LAND_HEIGHT_UP,                "WID_CS_FLAT_LAND_HEIGHT_UP");
+	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_FLAT_LAND_HEIGHT_DOWN,              "WID_GL_FLAT_LAND_HEIGHT_DOWN");
+	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_FLAT_LAND_HEIGHT_TEXT,              "WID_GL_FLAT_LAND_HEIGHT_TEXT");
+	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GL_FLAT_LAND_HEIGHT_UP,                "WID_GL_FLAT_LAND_HEIGHT_UP");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GP_PROGRESS_BAR,                       "WID_GP_PROGRESS_BAR");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GP_PROGRESS_TEXT,                      "WID_GP_PROGRESS_TEXT");
 	SQGSWindow.DefSQConst(engine, ScriptWindow::WID_GP_ABORT,                              "WID_GP_ABORT");
diff -r a048e90e006e -r 105cb37f0058 src/script/api/script_window.hpp
--- a/src/script/api/script_window.hpp	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/script/api/script_window.hpp	Tue Jul 17 17:01:20 2012 +0100
@@ -1220,22 +1220,20 @@
 		WID_GL_MAPSIZE_Y_PULLDOWN            = ::WID_GL_MAPSIZE_Y_PULLDOWN,            ///< Dropdown 'map Y size'.
 		WID_GL_TOWN_PULLDOWN                 = ::WID_GL_TOWN_PULLDOWN,                 ///< Dropdown 'No. of towns'.
 		WID_GL_INDUSTRY_PULLDOWN             = ::WID_GL_INDUSTRY_PULLDOWN,             ///< Dropdown 'No. of industries'.
-		WID_GL_RANDOM_EDITBOX                = ::WID_GL_RANDOM_EDITBOX,                ///< 'Random seed' editbox.
-		WID_GL_RANDOM_BUTTON                 = ::WID_GL_RANDOM_BUTTON,                 ///< 'Randomise' button.
 		WID_GL_GENERATE_BUTTON               = ::WID_GL_GENERATE_BUTTON,               ///< 'Generate' button.
 		WID_GL_START_DATE_DOWN               = ::WID_GL_START_DATE_DOWN,               ///< Decrease start year.
 		WID_GL_START_DATE_TEXT               = ::WID_GL_START_DATE_TEXT,               ///< Start year.
 		WID_GL_START_DATE_UP                 = ::WID_GL_START_DATE_UP,                 ///< Increase start year.
 		WID_GL_SNOW_LEVEL_DOWN               = ::WID_GL_SNOW_LEVEL_DOWN,               ///< Decrease snow level.
-		WID_GL_SNOW_LEVEL_TEXT               = ::WID_GL_SNOW_LEVEL_TEXT,               ///< Snow level.
+		WID_GL_SNOW_LEVEL_TEXT               = ::WID_GL_SNOW_LEVEL_TEXT,               ///< Snow line.
 		WID_GL_SNOW_LEVEL_UP                 = ::WID_GL_SNOW_LEVEL_UP,                 ///< Increase snow level.
-		WID_GL_TREE_PULLDOWN                 = ::WID_GL_TREE_PULLDOWN,                 ///< Dropdown 'Tree algorithm'.
-		WID_GL_LANDSCAPE_PULLDOWN            = ::WID_GL_LANDSCAPE_PULLDOWN,            ///< Dropdown 'Land generator'.
+		WID_GL_TREE_PULLDOWN                 = ::WID_GL_TREE_PULLDOWN,                 ///< Dropdown 'Trees'.
+		WID_GL_LANDSCAPE_PULLDOWN            = ::WID_GL_LANDSCAPE_PULLDOWN,            ///< Dropdown 'Generator'.
 		WID_GL_HEIGHTMAP_NAME_TEXT           = ::WID_GL_HEIGHTMAP_NAME_TEXT,           ///< Heightmap name.
 		WID_GL_HEIGHTMAP_SIZE_TEXT           = ::WID_GL_HEIGHTMAP_SIZE_TEXT,           ///< Size of heightmap.
 		WID_GL_HEIGHTMAP_ROTATION_PULLDOWN   = ::WID_GL_HEIGHTMAP_ROTATION_PULLDOWN,   ///< Dropdown 'Heightmap rotation'.
 		WID_GL_TERRAIN_PULLDOWN              = ::WID_GL_TERRAIN_PULLDOWN,              ///< Dropdown 'Terrain type'.
-		WID_GL_WATER_PULLDOWN                = ::WID_GL_WATER_PULLDOWN,                ///< Dropdown 'Sea level'.
+		WID_GL_WATER_PULLDOWN                = ::WID_GL_WATER_PULLDOWN,                ///< Dropdown 'Sea'.
 		WID_GL_RIVER_PULLDOWN                = ::WID_GL_RIVER_PULLDOWN,                ///< Dropdown 'Rivers'.
 		WID_GL_SMOOTHNESS_PULLDOWN           = ::WID_GL_SMOOTHNESS_PULLDOWN,           ///< Dropdown 'Smoothness'.
 		WID_GL_VARIETY_PULLDOWN              = ::WID_GL_VARIETY_PULLDOWN,              ///< Dropdown 'Variety distribution'.
@@ -1244,24 +1242,18 @@
 		WID_GL_WATER_NE                      = ::WID_GL_WATER_NE,                      ///< NE 'Water'/'Freeform'.
 		WID_GL_WATER_SE                      = ::WID_GL_WATER_SE,                      ///< SE 'Water'/'Freeform'.
 		WID_GL_WATER_SW                      = ::WID_GL_WATER_SW,                      ///< SW 'Water'/'Freeform'.
-	};
 
-	/** Widgets of the #CreateScenarioWindow class. */
-	enum CreateScenarioWidgets {
-		WID_CS_TEMPERATE                     = ::WID_CS_TEMPERATE,                     ///< Select temperate landscape style.
-		WID_CS_ARCTIC                        = ::WID_CS_ARCTIC,                        ///< Select arctic landscape style.
-		WID_CS_TROPICAL                      = ::WID_CS_TROPICAL,                      ///< Select tropical landscape style.
-		WID_CS_TOYLAND                       = ::WID_CS_TOYLAND,                       ///< Select toy-land landscape style.
-		WID_CS_EMPTY_WORLD                   = ::WID_CS_EMPTY_WORLD,                   ///< Generate an empty flat world.
-		WID_CS_RANDOM_WORLD                  = ::WID_CS_RANDOM_WORLD,                  ///< Generate random land button
-		WID_CS_MAPSIZE_X_PULLDOWN            = ::WID_CS_MAPSIZE_X_PULLDOWN,            ///< Pull-down arrow for x map size.
-		WID_CS_MAPSIZE_Y_PULLDOWN            = ::WID_CS_MAPSIZE_Y_PULLDOWN,            ///< Pull-down arrow for y map size.
-		WID_CS_START_DATE_DOWN               = ::WID_CS_START_DATE_DOWN,               ///< Decrease start year (start earlier).
-		WID_CS_START_DATE_TEXT               = ::WID_CS_START_DATE_TEXT,               ///< Clickable start date value.
-		WID_CS_START_DATE_UP                 = ::WID_CS_START_DATE_UP,                 ///< Increase start year (start later).
-		WID_CS_FLAT_LAND_HEIGHT_DOWN         = ::WID_CS_FLAT_LAND_HEIGHT_DOWN,         ///< Decrease flat land height.
-		WID_CS_FLAT_LAND_HEIGHT_TEXT         = ::WID_CS_FLAT_LAND_HEIGHT_TEXT,         ///< Clickable flat land height value.
-		WID_CS_FLAT_LAND_HEIGHT_UP           = ::WID_CS_FLAT_LAND_HEIGHT_UP,           ///< Increase flat land height.
+		WID_GL_DRIVING_SIDE_PULLDOWN         = ::WID_GL_DRIVING_SIDE_PULLDOWN,         ///< Dropdown 'drive on right/left side'.
+		WID_GL_TOWN_NAMES_PULLDOWN           = ::WID_GL_TOWN_NAMES_PULLDOWN,           ///< Dropdown 'Town names'.
+		WID_GL_ADVANCED_TEXT_PLUS_MIN        = ::WID_GL_ADVANCED_TEXT_PLUS_MIN,        ///< Button with icon "Advanced +/Advanced -".
+		WID_GL_NEW_MAP                       = ::WID_GL_NEW_MAP,                       ///< Button with icon "New Map"
+		WID_GL_SCENARIO                      = ::WID_GL_SCENARIO,                      ///< Button with icon "Load Scenaario"
+		WID_GL_HEIGHT_MAP                    = ::WID_GL_HEIGHT_MAP,                    ///< Button with icon "Load Height Map"
+		WID_GL_FLAT_LAND_HEIGHT_DOWN         = ::WID_GL_FLAT_LAND_HEIGHT_DOWN,         ///< Decrease flat land height..
+		WID_GL_FLAT_LAND_HEIGHT_TEXT         = ::WID_GL_FLAT_LAND_HEIGHT_TEXT,         ///< Clickable flat land height value..
+		WID_GL_FLAT_LAND_HEIGHT_UP           = ::WID_GL_FLAT_LAND_HEIGHT_UP,           ///< Increase flat land height..
+
+
 	};
 
 	/** Widgets of the #GenerateProgressWindow class. */
diff -r a048e90e006e -r 105cb37f0058 src/script/api/template/template_window.hpp.sq
--- a/src/script/api/template/template_window.hpp.sq	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/script/api/template/template_window.hpp.sq	Tue Jul 17 17:01:20 2012 +0100
@@ -79,8 +79,6 @@
 	template <> inline int Return<ScriptWindow::SaveLoadWidgets>(HSQUIRRELVM vm, ScriptWindow::SaveLoadWidgets res) { sq_pushinteger(vm, (int32)res); return 1; }
 	template <> inline ScriptWindow::GenerateLandscapeWidgets GetParam(ForceType<ScriptWindow::GenerateLandscapeWidgets>, HSQUIRRELVM vm, int index, SQAutoFreePointers *ptr) { SQInteger tmp; sq_getinteger(vm, index, &tmp); return (ScriptWindow::GenerateLandscapeWidgets)tmp; }
 	template <> inline int Return<ScriptWindow::GenerateLandscapeWidgets>(HSQUIRRELVM vm, ScriptWindow::GenerateLandscapeWidgets res) { sq_pushinteger(vm, (int32)res); return 1; }
-	template <> inline ScriptWindow::CreateScenarioWidgets GetParam(ForceType<ScriptWindow::CreateScenarioWidgets>, HSQUIRRELVM vm, int index, SQAutoFreePointers *ptr) { SQInteger tmp; sq_getinteger(vm, index, &tmp); return (ScriptWindow::CreateScenarioWidgets)tmp; }
-	template <> inline int Return<ScriptWindow::CreateScenarioWidgets>(HSQUIRRELVM vm, ScriptWindow::CreateScenarioWidgets res) { sq_pushinteger(vm, (int32)res); return 1; }
 	template <> inline ScriptWindow::GenerationProgressWidgets GetParam(ForceType<ScriptWindow::GenerationProgressWidgets>, HSQUIRRELVM vm, int index, SQAutoFreePointers *ptr) { SQInteger tmp; sq_getinteger(vm, index, &tmp); return (ScriptWindow::GenerationProgressWidgets)tmp; }
 	template <> inline int Return<ScriptWindow::GenerationProgressWidgets>(HSQUIRRELVM vm, ScriptWindow::GenerationProgressWidgets res) { sq_pushinteger(vm, (int32)res); return 1; }
 	template <> inline ScriptWindow::GoalListWidgets GetParam(ForceType<ScriptWindow::GoalListWidgets>, HSQUIRRELVM vm, int index, SQAutoFreePointers *ptr) { SQInteger tmp; sq_getinteger(vm, index, &tmp); return (ScriptWindow::GoalListWidgets)tmp; }
diff -r a048e90e006e -r 105cb37f0058 src/settings_type.h
--- a/src/settings_type.h	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/settings_type.h	Tue Jul 17 17:01:20 2012 +0100
@@ -216,12 +216,14 @@
 	Year   starting_year;                    ///< starting date
 	uint8  map_x;                            ///< X size of map
 	uint8  map_y;                            ///< Y size of map
+	bool   advanced_settings;                ///< advanced settings button in map generation screens
 	byte   land_generator;                   ///< the landscape generator
 	byte   oil_refinery_limit;               ///< distance oil refineries allowed from map edge
 	byte   snow_line_height;                 ///< a number 0-15 that configured snow line height
 	byte   tgen_smoothness;                  ///< how rough is the terrain from 0-3
 	byte   tree_placer;                      ///< the tree placer algorithm
 	byte   heightmap_rotation;               ///< rotation director for the heightmap
+	bool   terrain_type_totally_flat;        ///< the mountainousness of the landscape
 	byte   se_flat_world_height;             ///< land height a flat world gets in SE
 	byte   town_name;                        ///< the town name generator used for town names
 	byte   landscape;                        ///< the landscape we're currently in
diff -r a048e90e006e -r 105cb37f0058 src/table/settings.ini
--- a/src/table/settings.ini	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/table/settings.ini	Tue Jul 17 17:01:20 2012 +0100
@@ -1960,6 +1960,16 @@
 min      = 0
 max      = 16
 
+[SDT_BOOL]
+base     = GameSettings
+var      = game_creation.advanced_settings
+def      = false
+
+[SDT_BOOL]
+base     = GameSettings
+var      = game_creation.terrain_type_totally_flat
+def      = false
+
 [SDT_VAR]
 base     = GameSettings
 var      = game_creation.custom_town_number
diff -r a048e90e006e -r 105cb37f0058 src/terraform_gui.cpp
--- a/src/terraform_gui.cpp	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/terraform_gui.cpp	Tue Jul 17 17:01:20 2012 +0100
@@ -645,7 +645,7 @@
 
 			case WID_ETT_NEW_SCENARIO: // gen random land
 				this->HandleButtonClick(widget);
-				ShowCreateScenario();
+				StartScenarioEditor();
 				break;
 
 			case WID_ETT_RESET_LANDSCAPE: // Reset landscape
diff -r a048e90e006e -r 105cb37f0058 src/widgets/genworld_widget.h
--- a/src/widgets/genworld_widget.h	Mon Jul 02 19:31:26 2012 +0000
+++ b/src/widgets/genworld_widget.h	Tue Jul 17 17:01:20 2012 +0100
@@ -25,9 +25,6 @@
 	WID_GL_TOWN_PULLDOWN,               ///< Dropdown 'No. of towns'.
 	WID_GL_INDUSTRY_PULLDOWN,           ///< Dropdown 'No. of industries'.
 
-	WID_GL_RANDOM_EDITBOX,              ///< 'Random seed' editbox.
-	WID_GL_RANDOM_BUTTON,               ///< 'Randomise' button.
-
 	WID_GL_GENERATE_BUTTON,             ///< 'Generate' button.
 
 	WID_GL_START_DATE_DOWN,             ///< Decrease start year.
@@ -35,10 +32,10 @@
 	WID_GL_START_DATE_UP,               ///< Increase start year.
 
 	WID_GL_SNOW_LEVEL_DOWN,             ///< Decrease snow level.
-	WID_GL_SNOW_LEVEL_TEXT,             ///< Snow level.
+	WID_GL_SNOW_LEVEL_TEXT,             ///< Snow line.
 	WID_GL_SNOW_LEVEL_UP,               ///< Increase snow level.
 
-	WID_GL_TREE_PULLDOWN,               ///< Dropdown 'Tree algorithm'.
+	WID_GL_TREE_PULLDOWN,               ///< Dropdown 'Trees'.
 	WID_GL_LANDSCAPE_PULLDOWN,          ///< Dropdown 'Land generator'.
 
 	WID_GL_HEIGHTMAP_NAME_TEXT,         ///< Heightmap name.
@@ -46,7 +43,7 @@
 	WID_GL_HEIGHTMAP_ROTATION_PULLDOWN, ///< Dropdown 'Heightmap rotation'.
 
 	WID_GL_TERRAIN_PULLDOWN,            ///< Dropdown 'Terrain type'.
-	WID_GL_WATER_PULLDOWN,              ///< Dropdown 'Sea level'.
+	WID_GL_WATER_PULLDOWN,              ///< Dropdown 'Sea'.
 	WID_GL_RIVER_PULLDOWN,              ///< Dropdown 'Rivers'.
 	WID_GL_SMOOTHNESS_PULLDOWN,         ///< Dropdown 'Smoothness'.
 	WID_GL_VARIETY_PULLDOWN,            ///< Dropdown 'Variety distribution'.
@@ -56,24 +53,17 @@
 	WID_GL_WATER_NE,                    ///< NE 'Water'/'Freeform'.
 	WID_GL_WATER_SE,                    ///< SE 'Water'/'Freeform'.
 	WID_GL_WATER_SW,                    ///< SW 'Water'/'Freeform'.
-};
 
-/** Widgets of the #CreateScenarioWindow class. */
-enum CreateScenarioWidgets {
-	WID_CS_TEMPERATE,              ///< Select temperate landscape style.
-	WID_CS_ARCTIC,                 ///< Select arctic landscape style.
-	WID_CS_TROPICAL,               ///< Select tropical landscape style.
-	WID_CS_TOYLAND,                ///< Select toy-land landscape style.
-	WID_CS_EMPTY_WORLD,            ///< Generate an empty flat world.
-	WID_CS_RANDOM_WORLD,           ///< Generate random land button
-	WID_CS_MAPSIZE_X_PULLDOWN,     ///< Pull-down arrow for x map size.
-	WID_CS_MAPSIZE_Y_PULLDOWN,     ///< Pull-down arrow for y map size.
-	WID_CS_START_DATE_DOWN,        ///< Decrease start year (start earlier).
-	WID_CS_START_DATE_TEXT,        ///< Clickable start date value.
-	WID_CS_START_DATE_UP,          ///< Increase start year (start later).
-	WID_CS_FLAT_LAND_HEIGHT_DOWN,  ///< Decrease flat land height.
-	WID_CS_FLAT_LAND_HEIGHT_TEXT,  ///< Clickable flat land height value.
-	WID_CS_FLAT_LAND_HEIGHT_UP,    ///< Increase flat land height.
+	WID_GL_DRIVING_SIDE_PULLDOWN,       ///< Dropdown 'drive on right/left side'.
+	WID_GL_TOWN_NAMES_PULLDOWN,         ///< Dropdown 'Town names'.
+	WID_GL_ADVANCED_TEXT_PLUS_MIN,      ///< Button with icon "Advanced +/Advanced -".
+	WID_GL_NEW_MAP,                     ///< Button with icon "New Map".
+	WID_GL_SCENARIO,                    ///< Button with icon "Load Scenaario".
+	WID_GL_HEIGHT_MAP,                  ///< Button with icon "Load Height Map".
+	WID_GL_FLAT_LAND_HEIGHT_DOWN,       ///< Decrease flat land height.
+	WID_GL_FLAT_LAND_HEIGHT_TEXT,       ///< Clickable flat land height value.
+	WID_GL_FLAT_LAND_HEIGHT_UP,         ///< Increase flat land height.
+
 };
 
 /** Widgets of the #GenerateProgressWindow class. */
